### PR TITLE
Add secondary iface to KIND network

### PR DESCRIPTION
**Pull request by ysksuzuki**
Add secondary iface to KIND network, so that we can test the NodePort via secondary iface.

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Part of #18846

### CODE REVIEW

**Comment by ysksuzuki** [comment]
conformance-e2e on this PR has passed
https://github.com/cilium/cilium/actions/runs/5303659307

**Comment by ysksuzuki** [comment]
Tested the secondary if connectivity manually

```
// DSR + native routing + devices eth+(eth0, eth1)
cilium install --wait \
    --chart-directory=/home/yusuke/go/src/github.com/cilium/cilium/install/kubernetes/cilium \
    --helm-set=image.repository=quay.io/ysksuzuki/cilium-dev \
    --helm-set=image.useDigest=false \
    --helm-set=image.tag=main.0 \
    --helm-set=operator.image.repository=quay.io/ysksuzuki/operator \
    --helm-set=operator.image.suffix="" \
    --helm-set=operator.image.tag=main.0 \
    --helm-set=operator.image.useDigest=false \
    --helm-set=hubble.relay.image.repository=quay.io/ysksuzuki/hubble-relay \
    --helm-set=hubble.relay.image.tag=main.0 \
    --helm-set=hubble.relay.image.useDigest=false \
    --helm-set=hubble.eventBufferCapacity=65535 \
    --rollback=false \
    --config monitor-aggregation=none \
    --nodes-without-cilium=kind-worker3 \
    --helm-set-string=kubeProxyReplacement=strict \
    --helm-set-string=routingMode=native \
    --helm-set-string=autoDirectNodeRoutes=true \
    --helm-set-string=ipv4NativeRoutingCIDR=10.244.0.0/16 \
    --helm-set-string=ipv6NativeRoutingCIDR=fd00:10:244::/56 \
    --helm-set-string=loadBalancer.acceleration=testing-only \
    --helm-set-string=loadBalancer.mode=dsr \
    --helm-set-string=loadBalancer.dsrDispatch=opt \
    --helm-set=bpf.masquerade=true \
    --helm-set=enableIPv6Masquerade=false \
    --helm-set=debug.enabled=true \
    --helm-set=ipv6.enabled=true \
    --helm-set-string=devices=eth+

// SNAT + vxlan + devices eth+(eth0, eth1)
cilium install --wait \
    --chart-directory=/home/yusuke/go/src/github.com/cilium/cilium/install/kubernetes/cilium \
    --helm-set=image.repository=quay.io/ysksuzuki/cilium-dev \
    --helm-set=image.useDigest=false \
    --helm-set=image.tag=main.0 \
    --helm-set=operator.image.repository=quay.io/ysksuzuki/operator \
    --helm-set=operator.image.suffix="" \
    --helm-set=operator.image.tag=main.0 \
    --helm-set=operator.image.useDigest=false \
    --helm-set=hubble.relay.image.repository=quay.io/ysksuzuki/hubble-relay \
    --helm-set=hubble.relay.image.tag=main.0 \
    --helm-set=hubble.relay.image.useDigest=false \
    --helm-set=hubble.eventBufferCapacity=65535 \
    --rollback=false \
    --config monitor-aggregation=none \
    --nodes-without-cilium=kind-worker3 \
    --helm-set-string=kubeProxyReplacement=strict \
    --helm-set-string=routingMode=tunnel \
    --helm-set-string=loadBalancer.acceleration=testing-only \
    --helm-set-string=loadBalancer.mode=snat \
    --helm-set=bpf.masquerade=true \
    --helm-set=enableIPv6Masquerade=false \
    --helm-set=debug.enabled=true \
    --helm-set=ipv6.enabled=true \
    --helm-set-string=devices=eth+

$ docker exec kind-control-plane ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: cilium_net@cilium_host: <BROADCAST,MULTICAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether ee:59:13:9e:c4:9e brd ff:ff:ff:ff:ff:ff
    inet6 fe80::ec59:13ff:fe9e:c49e/64 scope link
       valid_lft forever preferred_lft forever
3: cilium_host@cilium_net: <BROADCAST,MULTICAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 6a:5b:ac:34:40:c4 brd ff:ff:ff:ff:ff:ff
    inet 10.244.0.28/32 scope global cilium_host
       valid_lft forever preferred_lft forever
    inet6 fd00:10:244::5532/64 scope global
       valid_lft forever preferred_lft forever
    inet6 fe80::685b:acff:fe34:40c4/64 scope link
       valid_lft forever preferred_lft forever
5: lxc_health@if4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 1a:d7:f2:97:93:a8 brd ff:ff:ff:ff:ff:ff link-netnsid 1
    inet6 fe80::18d7:f2ff:fe97:93a8/64 scope link
       valid_lft forever preferred_lft forever
28: eth0@if29: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 xdpgeneric/id:2426 qdisc noqueue state UP group default qlen 1000
    link/ether 02:42:ac:14:00:04 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 172.20.0.4/16 brd 172.20.255.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fc00:c111::4/64 scope global nodad
       valid_lft forever preferred_lft forever
    inet6 fe80::42:acff:fe14:4/64 scope link
       valid_lft forever preferred_lft forever
34: eth1@if35: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 xdpgeneric/id:2484 qdisc noqueue state UP group default qlen 1000
    link/ether 02:42:ac:15:00:03 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 172.21.0.3/16 brd 172.21.255.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fc00:c112::3/64 scope global nodad
       valid_lft forever preferred_lft forever
    inet6 fe80::42:acff:fe15:3/64 scope link
       valid_lft forever preferred_lft forever

$ kubectl  -n cilium-test get svc
NAME              TYPE       CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
echo-other-node   NodePort   10.96.11.5      <none>        8080:30476/TCP   8m24s
echo-same-node    NodePort   10.96.105.203   <none>        8080:31186/TCP   8m24s

// Secondary IPv4
$ docker exec kind-worker3 curl http://172.21.0.3:30476
<html>
  <head>
    <link
      rel="stylesheet"
      href="https://use.fontawesome.com/releases/v5.8.2/css/all.css"
      integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay"
      crossorigin="anonymous"

// Secondary IPv6
$ docker exec kind-worker3 curl http://[fc00:c112::3]:30476
<html>
  <head>
    <link
      rel="stylesheet"
      href="https://use.fontawesome.com/releases/v5.8.2/css/all.css"
      integrity="sha384-oS3vJWv+0UjzBfQzYUhtDYW+Pj2yciDJxpsK1OYPAYjqT085Qq/1cq5FLXAZQ7Ay"
      crossorigin="anonymous"

```

File: contrib/scripts/kind.sh [hunk 1]
```diff
@@ -150,15 +161,20 @@ kubeadmConfigPatches:
...
   if ! [ -f "${CILIUM_ROOT}/test/l4lb/bpf_xdp_veth_host.o" ]; then
     pushd "${CILIUM_ROOT}/test/l4lb/" > /dev/null
     clang -O2 -Wall -target bpf -c bpf_xdp_veth_host.c -o bpf_xdp_veth_host.o
     popd > /dev/null
   fi
 
-  for ifc in /sys/class/net/"${bridge_dev}"/brif/*; do
-    ifc="${ifc#"/sys/class/net/${bridge_dev}/brif/"}"
+  for ifc in /sys/class/net/"${bridge_dev}"*/brif/*; do
+    ifc=$(echo $ifc | sed "s:/sys/class/net/${bridge_dev}.*/brif/::")
```
Comment (bimmlerd) [inline contrib/scripts/kind.sh]: let's use `$SED` here, which should point to GNU sed on macos too

File: contrib/scripts/kind-down.sh [hunk 2]
```diff
@@ -11,9 +11,11 @@ fi
 
 default_cluster_name="kind"
 default_network="kind-cilium"
+default_secondary_network="${default_network}-secondary"
```
Comment (bimmlerd) [inline contrib/scripts/kind-down.sh]: maybe just `secondary` instead of `default_secondary`?

**Review by bimmlerd — CHANGES_REQUESTED** [review]
(sorry for the early review, didn't realise this was still in draft)

Should we be gating this behind a flag? I guess most people don't need a secondary interface in their dev cycle, and I see value in keeping the default as simple as possible.

Other than nits, implementation looks good though.

File: contrib/scripts/kind.sh (hunk 1 above)
Comment (ysksuzuki) [reply contrib/scripts/kind.sh]: Nice catch! Thanks

**Comment by ysksuzuki** [comment]
> Should we be gating this behind a flag? I guess most people don't need a secondary interface in their dev cycle, and I see value in keeping the default as simple as possible.

Make sense. I will add a flag, e.g. `--secondary`, and it set up a simple single-network by default.

**Review by bimmlerd — APPROVED** [review]

**Review by nbusseneau — APPROVED** [review]

**Comment by ysksuzuki** [comment]
Merging https://github.com/cilium/cilium/pull/26856 made the conflict. I have to rebase my branch.

**Comment by ysksuzuki** [comment]
Hi @brb , please take a look at this PR

File: .github/workflows/conformance-e2e.yaml [hunk 3]
```diff
@@ -369,7 +369,7 @@ jobs:
             if [ "${{ matrix.ipv6 }}" == "false" ]; then
               IP_FAM="ipv4"
             fi
-            ./contrib/scripts/kind.sh --xdp "" 3 "" "" "${{ matrix.kube-proxy }}" \$IP_FAM
+            ./contrib/scripts/kind.sh --xdp --secondary-interface "" 3 "" "" "${{ matrix.kube-proxy }}" \$IP_FAM
```
Comment (brb) [inline .github/workflows/conformance-e2e.yaml]: Maybe let's call it `--secondary-network`?

Comment (ysksuzuki) [reply .github/workflows/conformance-e2e.yaml]: Done

**Review by brb — APPROVED** [review]
Thanks!

**Pull request merged**